### PR TITLE
`@0x/contracts-integrations`: Seeded RNG and simulation logging

### PR DESCRIPTION
**Pull request by mzhu25**
## Description
Addresses some fuzzing pipeline needs, namely:
- a seeded RNG. Tries to use `process.env.UUID` as the seed, which should allow us to use https://hub.docker.com/r/andyneff/uuidgen/
- Logging in JSON to stderr. Will log each function assertion executed, including the arguments and  any custom `txData` that it was executed with. On failure, `Simulation` will log the full error and dump the state of the `SimulationEnvironment` at that point, before rethrowing the error. 

This PR should also fix the `FunctionAssertion` lifetime issue mentioned in https://github.com/0xProject/0x-monorepo/pull/2366#discussion_r349901928, though this hasn't been thoroughly tested

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/integrations/test/fuzz_tests/stake_management_test.ts [hunk 1]
```diff
@@ -26,13 +26,19 @@ export class StakeManagementSimulation extends Simulation {
...
-            const action = _.sample(actions);
+            const action = Pseudorandom.sample(actions);
             yield (await action!.next()).value; // tslint:disable-line:no-non-null-assertion
         }
     }
 }
 
-blockchainTests.skip('Stake management fuzz test', env => {
+blockchainTests('Stake management fuzz test', env => {
+    before(function(): void {
```
Comment (jalextowle) [inline contracts/integrations/test/fuzz_tests/stake_management_test.ts]: slick

File: contracts/integrations/test/framework/utils/logger.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+class Logger {
+    private _step = 0;
+
+    constructor() {
+        console.warn(
+            JSON.stringify({
+                level: 'info',
+                time: new Date(),
+                msg: `Pseudorandom seed: ${Pseudorandom.seed}`,
```
Comment (jalextowle) [inline contracts/integrations/test/framework/utils/logger.ts]: 🙏

File: contracts/integrations/test/framework/assertions/decreaseStakingPoolOperatorShare.ts [hunk 3]
```diff
@@ -17,21 +16,16 @@ export function validDecreaseStakingPoolOperatorShareAssertion(
...
-            after: async (_beforeInfo, _result: FunctionResult, args: [string, number], txData: Partial<TxData>) => {
-                const [poolId, expectedOperatorShare] = args;
+    return new FunctionAssertion<[string, number], {}, void>(stakingWrapper, 'decreaseStakingPoolOperatorShare', {
+        after: async (_beforeInfo, _result: FunctionResult, args: [string, number], _txData: Partial<TxData>) => {
+            const [poolId, expectedOperatorShare] = args;
 
-                logUtils.log(`decreaseStakingPoolOperatorShare(${poolId}, ${expectedOperatorShare})`);
+            // Checks that the on-chain pool's operator share has been updated.
+            const { operatorShare } = await stakingWrapper.getStakingPool(poolId).callAsync();
+            expect(operatorShare).to.bignumber.equal(expectedOperatorShare);
```
Comment (jalextowle) [inline contracts/integrations/test/framework/assertions/decreaseStakingPoolOperatorShare.ts]: These should eventually be replaced with logging, right?

Comment (jalextowle) [reply contracts/integrations/test/framework/assertions/decreaseStakingPoolOperatorShare.ts]: Do you just plan to do that in a separate PR?

Comment (mzhu25) [reply contracts/integrations/test/framework/assertions/decreaseStakingPoolOperatorShare.ts]: nah, if any of these `expect`s throw, the `Simulation` class will log the error in JSON format before re-throwing

File: contracts/integrations/test/framework/assertions/moveStake.ts [hunk 4]
```diff
@@ -86,61 +86,50 @@ export function validMoveStakeAssertion(
...
 
-                // Fetches on-chain global stake balances and checks against local balances
-                const globalUndelegatedStake = await stakingWrapper
-                    .getGlobalStakeByStatus(StakeStatus.Undelegated)
+            // Fetches on-chain pool stake balances and checks against local balances
+            for (const poolId of updatedPools) {
+                const stakeDelegatedByOwner = await stakingWrapper
+                    .getStakeDelegatedToPoolByOwner(owner, poolId)
                     .callAsync();
-                const globalDelegatedStake = await stakingWrapper
```
Comment (hysz) [inline contracts/integrations/test/framework/assertions/moveStake.ts]: How come we removed this check?

File: contracts/integrations/test/framework/utils/logger.ts (hunk 2 above)
Comment (hysz) [reply contracts/integrations/test/framework/utils/logger.ts]: 🙏

File: contracts/integrations/test/framework/assertions/moveStake.ts (hunk 4 above)
Comment (mzhu25) [reply contracts/integrations/test/framework/assertions/moveStake.ts]: it's there, just got moved up a few lines 🙂

**Pull request merged**